lib/hawkular/inventory/entities.rb

```diff
@@ -117,8 +108,8 @@ def metrics(recursive = false)
       children(recursive).collect(&:metrics).flat_map(&:itself).concat(@metrics)
     end
 
-    def metrics_by_type(type)
-      @metrics.select { |m| m.type == type }
+    def metrics_by_family(family)
+      @metrics.select { |m| m.family == family }
```

Comment:
What exactly a family is? Is akin to a meta type? e.g. servers (widlfly 10, eap 6 , eap 7, jdg) ?

Reply:
No, it's the "family name" in prometheus, for instance a metric with ID `jvm_memory_pool_bytes_used{host=locahost,feedid=1234}` has family name `jvm_memory_pool_bytes_used`.
I'm not sure yet if we're going to be able to get availability metrics from family names, I think it's something @lucasponce will face with the task on hawkular-provider.

Reply:
There are nothing called "availability metric" in the new inventory or Prometheus itself. At least, it isn't explicitly called that (i.e. there is not a first-class concept of an availability metric like we have in Hawkular Metrics). You just 'have to know' a specific metric corresponds to UP or DOWN. For example:

https://github.com/hawkular/hawkular-commons/blob/master/hawkular-inventory-parent/hawkular-inventory-service/src/main/resources/wildfly-10-jmx-exporter.yml#L72-L86

```
- pattern: "^jboss.as<deployment=(.+)><>status: OK"
    attrNameSnakeCase: true
    name: wildfly_deployment_availability
    value: 1
    type: GAUGE
    labels:
      name: $1
 - pattern: "^jboss.as<deployment=(.+)><>status: (FAILED|STOPPED)"
    attrNameSnakeCase: true
    name: wildfly_deployment_availability
    value: 0
    type: GAUGE
    labels:
      name: $1
```

This is a "gauge" metric in Prometheus. We just happen to define it to have a value of 1 if the deployment's status is "OK" and 0 if the deployment's status is "FAILED" or "STOPPED". You just have to know the gauge metric that we assign the family name "wildfly_deployment_availability" is really a 0 or 1 "availability" metric.